ddtrace/llmobs/utils.py

```diff
@@ -21,6 +24,15 @@
 Message = TypedDict("Message", {"content": str, "role": str}, total=False)
 
 
+class Prompt(BaseModel):
```

Comment:
I chose to use a pydantic model since it nicely handles a lot of the data model validation logic for us with built in error messages

And I think the object itself is a user-friendly display of the `Prompt` data model

Example:
When the `template` field is not a string after we called `Prompt(**some_json_object)`, the model will raise the following `ValidationError` which we can catch and display tot he user
```
Failed to parse prompt dictionary with validation error: ', 1 validation error for Prompt
template
  Input should be a valid string [type=string_type, input_value=1, input_type=int]
    For further information visit https://errors.pydantic.dev/2.8/v/string_type
```

Reply:
(Another example)
We can define a `PromptVariables` model to easily validate that the `variables` field of `Prompt` is a an arbitrary dictionary where keys and values are both strings
```
class PromptVariables(RootModel):
    root: Dict[str, Union[str, "PromptVariables"]]
```